openlane/steps/netgen.py

```diff
@@ -172,12 +183,64 @@ def run(self, state_in: State, **kwargs) -> Tuple[ViewsUpdate, MetricsUpdate]:
                     f"readnet spice {lib} 1",
                     file=f,
                 )
+        stats_file = os.path.join(reports_dir, "lvs.rpt")
+        stats_file_json = os.path.join(reports_dir, "lvs.json")
+        netgen_setup_script = os.path.join(get_script_dir(), "netgen", "setup.tcl")
+        mkdirp(reports_dir)
 
+        spice_files_commands = []
+        for lib in spice_files:
+            spice_files_commands.append(
+                f"puts \"Reading SPICE netlist file '{lib}'...\""
+            )
+            spice_files_commands.append(f"readnet spice {lib} $circuit2")
+
+        macros_commands = []
+        macros_commands += [
+            f"readnet verilog {state_in[DesignFormat.POWERED_NETLIST]} $circuit2"
+        ]
+
+        format_list = [
+            DesignFormat.POWERED_NETLIST,
+            DesignFormat.NETLIST,
```

Comment:
I think fallbacks are always going to cause issues and/or false positives here. Would it not be more prudent to simply require powered netlist and raise warnings if `NETGEN_INCLUDE_MACROS_NETLIST` is True?

Reply:
This allows blackboxing macros by including a Verilog header instead of a power netlist. `NETLIST` is a fallback because `POWERED_NETLIST` is a new addition to OpenLane and not commonly used. It was even suggested by Hosni to have SPICE models the highest priority.

Reply:
That explains `NETLIST` but doesn't exactly explain `VERILOG_HEADER`. Incidentally, why were the SPICE views not added?

Reply:
> That explains `NETLIST` but doesn't exactly explain `VERILOG_HEADER`. Incidentally, why were the SPICE views not added?

> This allows blackboxing macros by including a Verilog header instead of a power netlist. 

I don't have a test case for SPICE views yet.

Reply:
More on Verilog header: Netgen will blackbox a macro if the netlist is empty